cmd/cluster/cluster_create.go

```diff
@@ -0,0 +1,252 @@
+	ErrOutputTypeNotSupported = errors.New("output type is not supported. Choose type 'text' to use this feature")
+	ErrWritingSandboxSaveData = errors.New("unable to save sandbox data to file system")
+)
+
+type CreateOptions struct {
+	Context       context.Context
+	ArmoryClient  *configuration.ConfigClient
+	configuration *config.Configuration
+	contextNames  []string
+	credentials   *model.Credential
+	progressbar   *progressbar.ProgressBar
+	saveData      *model.SandboxClusterSaveData
+}
+
+func NewCreateClusterCmd(configuration *config.Configuration) *cobra.Command {
+	o := NewCreateOptions()
+
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{},
+		Short:   createClusterShort,
+		Long:    createClusterLong,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			o.InitializeConfiguration(configuration)
+			if err := o.Run(cmd); err != nil {
+				return err
+			}
+			return nil
+		},
+		SilenceUsage: true,
+	}
+	return cmd
+}
+
+func NewCreateOptions() *CreateOptions {
+	return &CreateOptions{}
+}
+
+// InitializeConfiguration will fetch Auth info to configure the REST client, so initialization must be deferred to runtime
+func (o *CreateOptions) InitializeConfiguration(cfg *config.Configuration) {
+	o.configuration = cfg
+	ac := configuration.NewClient(cfg)
+	o.ArmoryClient = ac
+	o.Context = o.ArmoryClient.ArmoryCloudClient.Context
+}
+
+// Run performs the execution of 'cluster create' sub command and saves the info for later use
+func (o *CreateOptions) Run(cmd *cobra.Command) error {
+	if o.configuration.GetOutputType() != output.Text {
+		return ErrOutputTypeNotSupported
+	}
+	ctx := cmd.Context()
+	agentPrefix := randomString(6)
+	credentials, err := o.ArmoryClient.Credentials().Create(ctx, o.createNamedCredential(agentPrefix))
+	if err != nil {
+		return err
+	}
+	environmentId := lo.If(lo.FromPtrOr[bool](o.configuration.GetIsTest(), false), "test-env").ElseF(o.configuration.GetCustomerEnvironmentId)
+	err = AssignCredentialRNARole(ctx, credentials, o.ArmoryClient, environmentId)
+	if err != nil {
+		return err
+	}
+	createSandboxRequest := o.createSandboxRequest(agentPrefix, credentials)
+	sandboxResponse, err := o.ArmoryClient.Sandbox().Create(ctx, createSandboxRequest)
+	if err != nil {
+		return err
+	}
+	o.InitializeProgressBar()
+	o.saveData = &model.SandboxClusterSaveData{
+		SandboxCluster:        model.SandboxCluster{},
+		CreateSandboxRequest:  *createSandboxRequest,
+		CreateSandboxResponse: *sandboxResponse,
+	}
+	for {
+		cluster, err := o.ArmoryClient.Sandbox().Get(ctx, o.saveData.CreateSandboxResponse.ClusterId)
+		if err != nil {
+			return err
+		}
+
+		done, err := o.UpdateProgressBar(cluster)
+		if err != nil {
+			return err
+		}
+
+		if done {
+			break
+		}
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) UpdateProgressBar(cluster *model.SandboxCluster) (bool, error) {
+	o.saveData.SandboxCluster = *cluster
+	o.progressbar.Describe(cluster.Status)
+	err := o.writeToSaveDataToSandboxFile()
```

Comment:
It looks like you're only using this for testing. I was surprised to find the file on my machine after trying it out locally, and especially surprised to see the clientID and secret in the file. Could we turn this off unless you're running tests?

Reply:
It's intentional. We likely do not need the clientID and secret stored and should not store it, but in order to use the cluster later, we need the clusterId and name of the agent at the very least if we want to use the cluster to get the preview cube config.